includes/Modules/Site_Verification.php

```diff
@@ -37,13 +37,62 @@
 final class Site_Verification extends Module implements Module_With_Scopes {
 	use Module_With_Scopes_Trait;
 
+	/**
+	 * Meta site verification type.
+	 */
+	const VERIFICATION_TYPE_META = 'META';
+
+	/**
+	 * File site verification type.
+	 */
+	const VERIFICATION_TYPE_FILE = 'FILE';
+
+	/**
+	 * Callback invoked after serving verification file.
+	 *
+	 * @var \Closure
+	 */
+	private $post_serve_verification_file_callback;
+
 	/**
 	 * Registers functionality through WordPress hooks.
 	 *
 	 * @since 1.0.0
 	 */
 	public function register() {
 		$this->register_scopes_hook();
+
+		add_action(
+			'admin_init',
+			function() {
+				$this->handle_verification_token();
+			}
+		);
+
+		$print_site_verification_meta = function() {
+			$this->print_site_verification_meta();
+		};
+
+		add_action( 'wp_head', $print_site_verification_meta );
+		add_action( 'login_head', $print_site_verification_meta );
+
+		// Exit handled by callback for testability.
+		$this->post_serve_verification_file_callback = function () {
+			exit;
+		};
```

Comment:
Why use this property and not directly call `exit` in the one place where the file is served?

Reply:
It's because we can't call `exit` during the test or it will kill the PHPUnit process.

See `Site_VerificationTest::test_file_verification`

I agree it's a bit odd, but I'm open to other ideas to handle this if you have any.

Reply:
Makes sense. I think we can solve this is another way that doesn't introduce a new property. Potentially we could for now just include an if-clause to not die based on one of the PHPUnit constants.

In the long run, we might need something like this in more places and could think about a better solution, potentially something similar to how WordPress has a `wp_die_handler` filter.

Reply:
> Potentially we could for now just include an if-clause to not die based on one of the PHPUnit constants.

I thought about this too and PHPUnit does not actually define any constants from what I could see. We could add one in our `phpunit.xml` configuration though. I also wasn't sure about how you felt adding in a conditional which was specific to testing.

> In the long run, we might need something like this in more places and could think about a better solution, potentially something similar to how WordPress has a `wp_die_handler` filter

I thought about this as well, and the problem with `wp_die` is that it renders html output by default, which would break the verification output.

Along those lines, we could do something like this:

```php
$exit = apply_filters( 'googlesitekit_exit_handler', 'exit' );
if ( is_callable( $exit ) ) {
    $exit();
}
```

Like you said, this would likely be useful in multiple places. Maybe we could create a new `Utils\Exit_Handler` class which includes this filter internally?

Then we could `( new Exit_Handler )->invoke()` or even `( new Exit_Handler )()` using `__invoke`. Thoughts?

Reply:
The `Utils\Exit_Handler` class sounds like a clean approach to me. 👍

Reply:
Added!